src/plugins/data/public/ui/filter_bar/filter_editor/lib/filter_editor_utils.ts

```diff
@@ -27,12 +29,14 @@ export function getFilterableFields(indexPattern: IIndexPattern) {
 
 export function getOperatorOptions(field: IFieldType) {
   return FILTER_OPERATORS.filter((operator) => {
-    return !operator.fieldTypes || operator.fieldTypes.includes(field.type);
+    if (operator.field) return operator.field(field);
+    if (operator.fieldTypes) return operator.fieldTypes.includes(field.type);
+    return true;
   });
 }
 
-export function validateParams(params: any, type: string) {
-  switch (type) {
+export function validateParams(params: any, field: IFieldType) {
```

Comment:
because we need to check against `field.esTypes` now, I am passing the whole field object